### PR TITLE
[cli] Fix EXPO_ROUTER_APP_ROOT being forced to app on windows

### PR DESCRIPTION
**Pull request by marklawlor**
# Why

Fix app loading and API routes for Expo Router apps being developed on Windows using `src/app`

# How

Changed `src/app` to `path.join('src', 'app')`

# Test Plan

Manually tested.

# Checklist

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: packages/@expo/cli/src/start/server/metro/instantiateMetro.ts [hunk 1]
```diff
@@ -307,7 +307,8 @@ function pruneCustomTransformOptions(
   }
 
   if (
-    transformOptions.customTransformOptions?.routerRoot &&
+    transformOptions.customTransformOptions &&
+    !transformOptions.customTransformOptions?.routerRoot &&
```
Comment (EvanBacon) [inline packages/@expo/cli/src/start/server/metro/instantiateMetro.ts:311]: as discussed on the call, we should normalize the filePath before matching.

**Comment by Garett-MacGowan** [comment]
Thanks. ETA? This is blocking me.

**Comment by Garett-MacGowan** [comment]
❤️

**Comment by brentvatne** [comment]
this is released now!

**Pull request merged**